clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -494,4 +495,53 @@ public static String readFileAsString(String path, Charset charset) throws IOExc
     public static String readFileAsString(String path) throws IOException {
         return Utils.readFileAsString(path, Charset.defaultCharset());
     }
+
+    /**
+     * Creates a set
+     * @param elems the elements
+     * @param <T> the type of element
+     * @return Set
+     */
+    public static <T> HashSet<T> mkSet(T... elems) {
+        HashSet<T> set = new HashSet<T>();
```

Comment:
@rhauch, I meant both. By passing the collection to the copy constructor of the `HashSet`, the initial size of the internal array is big enough to contain the elements of the collection. This avoids reallocation and rehashing. Note that `Arrays.asList` doesn't copy elements, it's just a view over the array. If this view is deemed too expensive (seems doubtful), we could keep the existing code, but then we should pass the correct sizing parameters to the `HashSet` constructor.